api/common/modelstatus.go

```diff
@@ -37,8 +45,40 @@ func (c *ModelStatusAPI) ModelStatus(tags ...names.ModelTag) ([]base.ModelStatus
 		return nil, err
 	}
 
-	results := make([]base.ModelStatus, len(result.Results))
-	for i, r := range result.Results {
+	if olderVersion {
+		// This caters for the fact that older version of base.ModelStatus
+		// did not have an Error field.
+		return c.processModelStatusResultsV3(result.Results)
+	}
+
+	return c.processModelStatusResults(result.Results)
+}
+
+func (c *ModelStatusAPI) processModelStatusResults(rs []params.ModelStatus) ([]base.ModelStatus, error) {
+	results := make([]base.ModelStatus, len(rs))
+	for i, r := range rs {
+		if r.Error != nil {
+			results[i].Error = errors.Trace(r.Error)
```

Comment:
we don't really need the trace here as the error is coming from the api call and there's little/no value in adding context which isn't really there

Reply:
Very specifically, this exact statement is where we translate v3 into v4. Any global error gets translated into a per-object error.
Which means that everything above the 'api' layer can just assume that result[i].Error has been set.

Reply:
I am agreeing with @jameinel. This has to stay.